pkg/utils/errors/multierror.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+)
+
+// NewErrorFormatFuncWithPrefix creates a new multierror.ErrorFormatFunc which can be used as an ErrorFormat on
+// multierror.Error instances. The error string is prefixed with <prefix>, all errors are concatenated at the end.
+// This is similar to multierror.ListFormatFunc but does not use any escape sequences, which will look weird in
+// the status of Kubernetes objects or controller logs.
+func NewErrorFormatFuncWithPrefix(prefix string) multierror.ErrorFormatFunc {
+	return func(es []error) string {
+		if len(es) == 1 {
+			return fmt.Sprintf("%s: 1 error occurred: %s", prefix, es[0])
+		}
+
+		errStrings := make([]string, len(es))
```

Comment:
What's the purpose/benefit of copying the errors over into `[]string` in order to iterate again over it when calling `strings.Join()`?
You are always iterating, can't you just construct the error message and return it?

Also, a small unit test could be added.

Reply:
Sorry, leftover.
Also added a small unit test, PTAL :)